enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/ParameterSettings.tsx

```diff
@@ -0,0 +1,114 @@
+import { useDebouncedCallback } from "@mantine/hooks";
+import { useCallback, useMemo } from "react";
+import { P, match } from "ts-pattern";
+import { t } from "ttag";
+
+import { Stack, Text, TextInput } from "metabase/ui";
+
+import { useSdkIframeEmbedSetupContext } from "../context";
+import { useHideParameter } from "../hooks";
+
+import { ParameterVisibilityToggle } from "./ParameterVisibilityToggle";
+
+export const ParameterSettings = () => {
+  const {
+    experience,
+    settings,
+    updateSettings,
+    availableParameters,
+    isLoadingParameters,
+  } = useSdkIframeEmbedSetupContext();
+
+  const { isParameterHidden, toggleParameterVisibility } = useHideParameter();
+
+  const isQuestionOrDashboardEmbed =
+    !!settings.questionId || !!settings.dashboardId;
+
+  const updateInitialParameterValue = useDebouncedCallback(
+    useCallback(
+      (paramId: string, value: string) => {
+        if (settings.dashboardId) {
+          updateSettings({
+            initialParameters: {
+              ...settings.initialParameters,
+              [paramId]: value,
+            },
+          });
+        } else if (settings.questionId) {
+          updateSettings({
+            initialSqlParameters: {
+              ...settings.initialSqlParameters,
+              [paramId]: value,
+            },
+          });
+        }
+      },
+      [settings, updateSettings],
+    ),
+    500,
+  );
+
+  const parameterValues = useMemo(() => {
+    if (settings.dashboardId) {
+      return settings.initialParameters;
+    } else if (settings.questionId) {
+      return settings.initialSqlParameters;
+    }
+
+    return {};
+  }, [settings]);
+
+  // Only show parameters for dashboards and questions
+  if (!isQuestionOrDashboardEmbed) {
+    return null;
+  }
+
+  if (isLoadingParameters) {
+    return (
+      <Text size="sm" c="text-medium">
+        {t`Loading parameters...`}
+      </Text>
+    );
+  }
+
+  if (availableParameters.length > 0) {
+    return (
+      <Stack>
+        {availableParameters.map((param) => {
+          const placeholder = match(param.default)
+            .with(P.array(), (arr) => arr.join(", "))
+            .with(P.nullish, () => param.name.toLowerCase())
+            .otherwise((val) => val.toString());
```

Comment:
When will the `otherwise` option be used? It doesn't seem I make this work. Can you provide testing step for that?

Reply:
I always got `param.default` as an array too, fixed in https://github.com/metabase/metabase/pull/60233/commits/23d28a099c351113bddf79d6794459feb4037616 where I simplified this placeholder logic.